contrib/wireshark/ucx_le64.lua

```diff
@@ -0,0 +1,1683 @@
+    local addr = tvbuf(16, 8)
+    local size = tvbuf(24, 8)
+    local offset = tvbuf(32, 8)
+    local off = 40
+
+    local msg = "RTR"
+    pinfo.cols.protocol = "UCX_RNDV"
+    tree = tree:add(ucp_rndv_proto, tvbuf)
+
+    local urf = ucp_rndv_fields
+    tree:add_le(urf.sreq_id, sreq_id)
+    tree:add_le(urf.rreq_id, rreq_id)
+    tree:add_le(urf.addr, addr)
+    tree:add_le(urf.size, size)
+    tree:add_le(urf.offset, offset)
+
+    if addr:le_uint64() ~= 0 then
+        local rkey = tvbuf(off, tvbuf:len() - off)
+        tree:add_le(urf.rkey, rkey)
+    end
+
+    msg = msg .. ", SReq: " .. hex_64(sreq_id:le_uint64():tohex())
+    msg = msg .. ", RReq: " .. hex_64(rreq_id:le_uint64():tohex())
+    msg = msg .. ", Addr: " .. hex_64(addr:le_uint64():tohex())
+    msg = msg .. ", Size: " .. size:le_uint64()
+    msg = msg .. ", Off: " .. offset:le_uint64()
+    tree:append_text(", " .. msg)
+    pinfo.cols.info = msg
+end
+
+function ucp_rndv_proto_dissect_rts(tvbuf, pinfo, tree)
+    local hdr = tvbuf(0, 8)
+    local ep_id = tvbuf(8, 8)
+    local req_id = tvbuf(16, 8)
+    local addr = tvbuf(24, 8)
+    local size = tvbuf(32, 8)
+    local op_code = tvbuf(40, 1)
+    local off = 41
+
+    local msg = "RTS"
+    pinfo.cols.protocol = "UCX_RNDV"
+    tree = tree:add(ucp_rndv_proto, tvbuf)
+
+    local urf = ucp_rndv_fields
+
+    local hdr_len = nil
+    if op_code:uint() == RNDV_AM then
+        local am_id = tvbuf(0, 2)
+        local flags = tvbuf(2, 2)
+        hdr_len = tvbuf(4, 4)
+
+        tree:add_le(urf.am_id, am_id)
+        tree:add_le(urf.reply, flags)
+        tree:add_le(urf.eager, flags)
+        tree:add_le(urf.rndv, flags)
+        tree:add_le(urf.nbx_reg, flags)
+        tree:add_le(urf.hdr_len, hdr_len)
+
+        msg = msg .. ", AM: " .. am_id:le_uint()
+        msg = msg .. ", Hdr: " .. hdr_len:le_uint()
+    else
+        tree:add_le(urf.hdr, hdr)
+    end
+
+    tree:add_le(urf.ep_id, ep_id)
+    tree:add_le(urf.req_id, req_id)
+    tree:add_le(urf.addr, addr)
+    tree:add_le(urf.size, size)
+    tree:add_le(urf.op_code, op_code)
+
+    local rkey_len = tvbuf:len() - off - hdr_len:le_uint()
+    if rkey_len > 0 then
+        local rkey = tvbuf(off, rkey_len)
+        tree:add(urf.rkey, rkey)
+    end
+
+    if hdr_len ~= nil and hdr_len:uint() > 0 then
+        local user = tvbuf(off + rkey_len, hdr_len:le_uint())
+        tree:add(urf.user_hdr, user)
+    end
+
+    msg = msg .. ", Ep: " .. hex_64(ep_id:le_uint64():tohex())
+    msg = msg .. ", Req: " .. hex_64(req_id:le_uint64():tohex())
+    msg = msg .. ", Size: " .. size:le_uint64()
+    msg = msg .. ", Rkey: " .. rkey_len
+    msg = msg .. ", Memory: " .. hex_64(addr:le_uint64():tohex())
+    tree:append_text(", " .. msg)
+    pinfo.cols.info = msg
+end
+
+
+local amo_sw_opcode = {
+    [0] = "ADD",
+    [1] = "AND",
+    [2] = "OR",
+    [3] = "XOR",
+    [4] = "SWAP",
+    [5] = "CSWAP",
+}
+
+local ucp_amo_sw_proto = Proto("UCX_AMO_SW", "UCX Atomic SW")
```

Comment:
maybe better highlight that it is UC**P** SW AMO, because we have atomic interfaces in both UCT and UCP libs

Reply:
Makes sense, renamed to UCP SW AMO.